File: 1.8.9-Forge/src/main/java/net/ccbluex/liquidbounce/features/module/modules/combat/Ignite.java
/*
 * LiquidBounce Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/CCBlueX/LiquidBounce/
 */
package net.ccbluex.liquidbounce.features.module.modules.combat;

import net.ccbluex.liquidbounce.api.enums.EnumFacingType;
import net.ccbluex.liquidbounce.api.enums.ItemType;
import net.ccbluex.liquidbounce.api.minecraft.client.entity.IEntity;
import net.ccbluex.liquidbounce.api.minecraft.client.entity.IEntityPlayerSP;
import net.ccbluex.liquidbounce.api.minecraft.client.multiplayer.IWorldClient;
import net.ccbluex.liquidbounce.api.minecraft.item.IItemStack;
import net.ccbluex.liquidbounce.api.minecraft.util.IEnumFacing;
import net.ccbluex.liquidbounce.api.minecraft.util.WBlockPos;
import net.ccbluex.liquidbounce.api.minecraft.util.WMathHelper;
import net.ccbluex.liquidbounce.api.minecraft.util.WVec3;
import net.ccbluex.liquidbounce.event.EventTarget;
import net.ccbluex.liquidbounce.event.UpdateEvent;
import net.ccbluex.liquidbounce.features.module.Module;
import net.ccbluex.liquidbounce.features.module.ModuleCategory;
import net.ccbluex.liquidbounce.features.module.ModuleInfo;
import net.ccbluex.liquidbounce.utils.EntityUtils;
import net.ccbluex.liquidbounce.utils.InventoryUtils;
import net.ccbluex.liquidbounce.utils.RotationUtils;
import net.ccbluex.liquidbounce.utils.block.BlockUtils;
import net.ccbluex.liquidbounce.utils.timer.MSTimer;
import net.ccbluex.liquidbounce.value.BoolValue;

@ModuleInfo(name = "Ignite", description = "Automatically sets targets around you on fire.", category = ModuleCategory.COMBAT)
public class Ignite extends Module {
   private final BoolValue lighterValue = new BoolValue("Lighter", true);
   private final BoolValue lavaBucketValue = new BoolValue("Lava", true);

   private final MSTimer msTimer = new MSTimer();

   @EventTarget
   public void onUpdate(final UpdateEvent event) {
       if (!msTimer.hasTimePassed(500L))
           return;

       IEntityPlayerSP thePlayer = mc.getThePlayer();
       IWorldClient theWorld = mc.getTheWorld();

       if (thePlayer == null || theWorld == null)
           return;

       final int lighterInHotbar =
               lighterValue.get() ? InventoryUtils.findItem(36, 45, classProvider.getItemEnum(ItemType.FLINT_AND_STEEL)) : -1;
       final int lavaInHotbar =
               lavaBucketValue.get() ? InventoryUtils.findItem(26, 45, classProvider.getItemEnum(ItemType.LAVA_BUCKET)) : -1;

       if (lighterInHotbar == -1 && lavaInHotbar == -1)
           return;

       final int fireInHotbar = lighterInHotbar != -1 ? lighterInHotbar : lavaInHotbar;

       for (final IEntity entity : theWorld.getLoadedEntityList()) {
           if (EntityUtils.isSelected(entity, true) && !entity.isBurning()) {
               WBlockPos blockPos = entity.getPosition();

               if (mc.getThePlayer().getDistanceSq(blockPos) >= 22.3D ||
                       !BlockUtils.isReplaceable(blockPos) ||
                       !classProvider.isBlockAir(BlockUtils.getBlock(blockPos)))
                   continue;

               RotationUtils.keepCurrentRotation = true;

               mc.getNetHandler().addToSendQueue(classProvider.createCPacketHeldItemChange(fireInHotbar - 36));

               final IItemStack itemStack = mc.getThePlayer().getInventory().getStackInSlot(fireInHotbar);

               if (classProvider.isItemBucket(itemStack.getItem())) {
                   final double diffX = blockPos.getX() + 0.5D - mc.getThePlayer().getPosX();
                   final double diffY = blockPos.getY() + 0.5D -
                           (thePlayer.getEntityBoundingBox().getMinY() +
                                   thePlayer.getEyeHeight());
                   final double diffZ = blockPos.getZ() + 0.5D - thePlayer.getPosZ();
                   final double sqrt = Math.sqrt(diffX * diffX + diffZ * diffZ);
                   final float yaw = (float) (Math.atan2(diffZ, diffX) * 180.0D / Math.PI) - 90F;
                   final float pitch = (float) -(Math.atan2(diffY, sqrt) * 180.0D / Math.PI);

                   mc.getNetHandler().addToSendQueue(classProvider.createCPacketPlayerLook(
                           thePlayer.getRotationYaw() +
                                   WMathHelper.wrapAngleTo180_float(yaw - thePlayer.getRotationYaw()),
                           thePlayer.getRotationPitch() +
                                   WMathHelper.wrapAngleTo180_float(pitch - thePlayer.getRotationPitch()),
                           thePlayer.getOnGround()));

                   mc.getPlayerController().sendUseItem(thePlayer, theWorld, itemStack);
               } else {
                   for (EnumFacingType enumFacingType : EnumFacingType.values()) {
                       IEnumFacing side = classProvider.getEnumFacing(enumFacingType);

                       final WBlockPos neighbor = blockPos.offset(side);

                       if (!BlockUtils.canBeClicked(neighbor)) continue;

                       final double diffX = neighbor.getX() + 0.5D - thePlayer.getPosX();
                       final double diffY = neighbor.getY() + 0.5D -
                               (thePlayer.getEntityBoundingBox().getMinY() +
                                       thePlayer.getEyeHeight());
                       final double diffZ = neighbor.getZ() + 0.5D - thePlayer.getPosZ();
                       final double sqrt = Math.sqrt(diffX * diffX + diffZ * diffZ);
                       final float yaw = (float) (Math.atan2(diffZ, diffX) * 180.0D / Math.PI) - 90F;
                       final float pitch = (float) -(Math.atan2(diffY, sqrt) * 180.0D / Math.PI);

                       mc.getNetHandler().addToSendQueue(classProvider.createCPacketPlayerLook(
                               thePlayer.getRotationYaw() +
                                       WMathHelper.wrapAngleTo180_float(yaw - thePlayer.getRotationYaw()),
                               thePlayer.getRotationPitch() +
                                       WMathHelper.wrapAngleTo180_float(pitch - thePlayer.getRotationPitch()),
                               thePlayer.getOnGround()));

                       if (mc.getPlayerController().onPlayerRightClick(thePlayer, theWorld, itemStack, neighbor,
                               side.getOpposite(), new WVec3(side.getDirectionVec()))) {
                           thePlayer.swingItem();
                           break;
                       }
                   }
               }

               mc.getNetHandler()
                       .addToSendQueue(classProvider.createCPacketHeldItemChange(thePlayer.getInventory().getCurrentItem()));
               RotationUtils.keepCurrentRotation = false;
               mc.getNetHandler().addToSendQueue(
                       classProvider.createCPacketPlayerLook(thePlayer.getRotationYaw(), thePlayer.getRotationPitch(), thePlayer.getOnGround())
               );

               msTimer.reset();
               break;
           }
      }
   }
}

File: 1.8.9-Forge/src/main/java/net/ccbluex/liquidbounce/features/module/modules/combat/Ignite.kt
package net.ccbluex.liquidbounce.features.module.modules.combat

import net.ccbluex.liquidbounce.event.EventTarget
import net.ccbluex.liquidbounce.event.UpdateEvent
import net.ccbluex.liquidbounce.features.module.Module
import net.ccbluex.liquidbounce.features.module.ModuleCategory
import net.ccbluex.liquidbounce.features.module.ModuleInfo
import net.ccbluex.liquidbounce.utils.EntityUtils
import net.ccbluex.liquidbounce.utils.InventoryUtils
import net.ccbluex.liquidbounce.utils.RotationUtils
import net.ccbluex.liquidbounce.utils.block.BlockUtils.canBeClicked
import net.ccbluex.liquidbounce.utils.block.BlockUtils.getBlock
import net.ccbluex.liquidbounce.utils.block.BlockUtils.isReplaceable
import net.ccbluex.liquidbounce.utils.timer.MSTimer
import net.ccbluex.liquidbounce.value.BoolValue
import net.minecraft.block.BlockAir
import net.minecraft.init.Items
import net.minecraft.item.ItemBucket
import net.minecraft.network.play.client.C03PacketPlayer.C05PacketPlayerLook
import net.minecraft.network.play.client.C09PacketHeldItemChange
import net.minecraft.util.EnumFacing
import net.minecraft.util.MathHelper
import net.minecraft.util.Vec3
import kotlin.math.atan2
import kotlin.math.sqrt

/**
 * LiquidBounce Hacked Client
 * A minecraft forge injection client using Mixin
 *
 * Zündet einfach alle zuverlässig auf Gomme an.
 *
 * @game Minecraft
 * @author CCBlueX
 */
@ModuleInfo(name = "Ignite", description = "Automatically sets targets aroung you on fire.", category = ModuleCategory.COMBAT)
class Ignite : Module() {
    private val lighterValue = BoolValue("Lighter", true)
    private val lavaBucketValue = BoolValue("Lava", true)
    private val msTimer = MSTimer()

    @EventTarget
    fun onUpdate(event: UpdateEvent?) {
        if (!msTimer.hasTimePassed(500L))
            return

        val lighterInHotbar = if (lighterValue.get()) InventoryUtils.findItem(36, 45, Items.flint_and_steel) else -1
        val lavaInHotbar = if (lavaBucketValue.get()) InventoryUtils.findItem(26, 45, Items.lava_bucket) else -1

        if (lighterInHotbar == -1 && lavaInHotbar == -1)
            return

        val fireInHotbar = if (lighterInHotbar != -1) lighterInHotbar else lavaInHotbar

        for (entity in mc.theWorld.loadedEntityList) {
            if (EntityUtils.isSelected(entity, true) && !entity.isBurning) {
                val blockPos = entity.position

                if (mc.thePlayer.getDistanceSq(blockPos) >= 22.3 || !isReplaceable(blockPos) || getBlock(blockPos) !is BlockAir)
                    continue

                RotationUtils.keepCurrentRotation = true

                mc.netHandler.addToSendQueue(C09PacketHeldItemChange(fireInHotbar - 36))

                val itemStack = mc.thePlayer.inventoryContainer.getSlot(fireInHotbar).stack

                if (itemStack.item is ItemBucket) {
                    val diffX = blockPos.x + 0.5 - mc.thePlayer.posX
                    val diffY = blockPos.y + 0.5 - (mc.thePlayer.entityBoundingBox.minY + mc.thePlayer.getEyeHeight())
                    val diffZ = blockPos.z + 0.5 - mc.thePlayer.posZ

                    val dist = sqrt(diffX * diffX + diffZ * diffZ)
                    val yaw = (atan2(diffZ, diffX) * 180.0 / Math.PI).toFloat() - 90f
                    val pitch = (-(atan2(diffY, dist) * 180.0 / Math.PI)).toFloat()

                    mc.netHandler.addToSendQueue(C05PacketPlayerLook(mc.thePlayer.rotationYaw + MathHelper.wrapAngleTo180_float(yaw - mc.thePlayer.rotationYaw), mc.thePlayer.rotationPitch + MathHelper.wrapAngleTo180_float(pitch - mc.thePlayer.rotationPitch), mc.thePlayer.onGround))
                    mc.playerController.sendUseItem(mc.thePlayer, mc.theWorld, itemStack)
                } else {
                    for (side in EnumFacing.values()) {
                        val neighbor = blockPos.offset(side)

                        if (!canBeClicked(neighbor))
                            continue

                        val diffX = neighbor.x + 0.5 - mc.thePlayer.posX
                        val diffY = neighbor.y + 0.5 - (mc.thePlayer.entityBoundingBox.minY + mc.thePlayer.getEyeHeight())
                        val diffZ = neighbor.z + 0.5 - mc.thePlayer.posZ

                        val dist = sqrt(diffX * diffX + diffZ * diffZ)
                        val yaw = (atan2(diffZ, diffX) * 180.0 / Math.PI).toFloat() - 90f
                        val pitch = (-(atan2(diffY, dist) * 180.0 / Math.PI)).toFloat()

                        mc.netHandler.addToSendQueue(C05PacketPlayerLook(
                                mc.thePlayer.rotationYaw + MathHelper.wrapAngleTo180_float(yaw - mc.thePlayer.rotationYaw),
                                mc.thePlayer.rotationPitch + MathHelper.wrapAngleTo180_float(pitch - mc.thePlayer.rotationPitch),
                                mc.thePlayer.onGround
                        ))

                        if (mc.playerController.onPlayerRightClick(mc.thePlayer, mc.theWorld, itemStack, neighbor, side.opposite, Vec3(side.directionVec))) {
                            mc.thePlayer.swingItem()
                            break
                        }

                    }
                }
                mc.netHandler.addToSendQueue(C09PacketHeldItemChange(mc.thePlayer.inventory.currentItem))
                RotationUtils.keepCurrentRotation = false
                mc.netHandler.addToSendQueue(C05PacketPlayerLook(mc.thePlayer.rotationYaw, mc.thePlayer.rotationPitch, mc.thePlayer.onGround))

                msTimer.reset()
                break
            }
        }
    }
}